timeseries/src/autogluon/timeseries/models/chronos/pipeline/utils.py

```diff
@@ -241,38 +376,99 @@ def cache_model_from_s3(s3_uri: str, force=False):
     return str(bucket_cache_path / prefix)
 
 
-class ChronosInferenceDataset:
+class ChronosInferenceDataset(IterableDataset):
```

Comment:
Do you think it's feasible to have just one `ChronosDataset` class? Currently it feels like there is a lot of duplication between the two classes. We could set `inference=True/False` in the constructor, which will only affect instance splitter - choosing between returning the last `context_length` observations from each series (+ future known covariates) if `inference=True`, and doing random slicing during training when `inference=False`.